doc/source/how-to-run-simulations.rst

```diff
@@ -1,291 +1,338 @@
 
-    We are actively working on these fronts so to make it trivial to run any FL workload
-    with Flower simulation.
-
-The current VCE allows you to run Federated Learning workloads in simulation mode
-whether you are prototyping simple scenarios on your personal laptop or you want to
-train a complex FL pipeline across multiple high-performance GPU nodes. While we add
-more capabilities to the VCE, the points below highlight some of the considerations to
-keep in mind when designing your FL pipeline with Flower. We also highlight a couple of
-current limitations in our implementation.
-
-GPU resources
-~~~~~~~~~~~~~
-
-The VCE assigns a share of GPU memory to a client that specifies the key ``num_gpus`` in
-``client_resources``. This being said, Ray (used internally by the VCE) is by default:
-
-- not aware of the total VRAM available on the GPUs. This means that if you set
-  ``num_gpus=0.5`` and you have two GPUs in your system with different (e.g. 32GB and
-  8GB) VRAM amounts, they both would run 2 clients concurrently.
-- not aware of other unrelated (i.e. not created by the VCE) workloads are running on
-  the GPU. Two takeaways from this are:
-
-  - Your Flower server might need a GPU to evaluate the `global model` after aggregation
-    (by instance when making use of the `evaluate method
-    <how-to-implement-strategies.html#the-evaluate-method>`_)
-  - If you want to run several independent Flower simulations on the same machine you
-    need to mask-out your GPUs with ``CUDA_VISIBLE_DEVICES="<GPU_IDs>"`` when launching
-    your experiment.
-
-In addition, the GPU resource limits passed to ``client_resources`` are not `enforced`
-(i.e. they can be exceeded) which can result in the situation of client using more VRAM
-than the ratio specified when starting the simulation.
-
-TensorFlow with GPUs
-++++++++++++++++++++
-
-When `using a GPU with TensorFlow <https://www.tensorflow.org/guide/gpu>`_ nearly your
-entire GPU memory of all your GPUs visible to the process will be mapped. This is done
-by TensorFlow for optimization purposes. However, in settings such as FL simulations
-where we want to split the GPU into multiple `virtual` clients, this is not a desirable
-mechanism. Luckily we can disable this default behavior by `enabling memory growth
-<https://www.tensorflow.org/guide/gpu#limiting_gpu_memory_growth>`_.
-
-This would need to be done in the main process (which is where the server would run) and
-in each Actor created by the VCE. By means of ``actor_kwargs`` we can pass the reserved
-key `"on_actor_init_fn"` in order to specify a function to be executed upon actor
-initialization. In this case, to enable GPU growth for TF workloads. It would look as
-follows:
+.. dropdown:: Are the CPU/GPU resources set for each ``ClientApp`` enforced?
```

Comment:
```suggestion
.. dropdown:: Are the CPU/GPU resources set for each ``ClientApp`` enforced?
```

I'm unclear what you mean here by enforce? Do you mean a greedy allocation of resources by the backend?

Reply:
By "enforce" i mean that even though you may choose `num_gpus=0.25` the simulation engine doesn't restrict the `ClientApp` from using a larger portion of the available VRAM. Earlier in the doc page I use the term _soft_ assignment of resources to convey the same idea. Maybe some other term other than "enforce" should be used?

Reply:
I've tweaked the question. I think it's clearer now. Thanks for flagging!